src/axis.jl

```diff
@@ -2,10 +2,47 @@ import Base.length
 import Base.iterate
 import Base.getindex
 import Base.lastindex
+import Base.to_index
+import Base.hash
+import Base.isequal
 
-struct Axis{T, A <: AbstractArray{T}}
-    name::Symbol # Name of the axis (as declared in the macro)
-    container::A
+struct AxisId{Name, T}
+    indice::T
+end
+
+function name(i::AxisId{Name,T}) where {Name,T}
+    return Name
+end
+
+function indice(i::AxisId{Name,T})::T where {Name,T}
+    return i.indice
+end
+
+Base.hash(i::AxisId, h::UInt) = hash(i.indice, h)
+
+# Permit the access to the entry of an array using an AxisId.
+Base.to_index(i::AxisId) = i.indice
+
+# Allow matching of AxisId key using the value of field indice (dict.jl:289)
+# and vice-versa
+Base.isequal(i::T, j::AxisId{N,T}) where {N,T} = isequal(i, j.indice)
+Base.isequal(i::AxisId{N,T}, j::T) where {N,T} = isequal(i.indice, j)
+
+iterate(i::AxisId) = (i, nothing)
+iterate(i::AxisId, ::Any) = nothing
```

Comment:
This means you cannot have a vector of `AxisId`?

Reply:
It means that you can iterate over an `AxisId` but its a singleton containing only the `AxisId`.
```julia
for i in axisid
 # i is the axisid
end
```
It's like Number.